pkg/pdatatest/ptracetest/traces.go

```diff
@@ -301,3 +297,84 @@ func CompareSpan(expected, actual ptrace.Span) error {
 
 	return errs
 }
+
+// CompareEvents compares each part of two given SpanEventSlice and returns
+// an error if they don't match. The error describes what didn't match.
+func CompareEvents(expected, actual ptrace.SpanEventSlice) (errs error) {
+	if expected.Len() != actual.Len() {
+		errs = multierr.Append(errs, fmt.Errorf("number of events doesn't match expected: %d, actual: %d",
+			expected.Len(), actual.Len()))
+		return errs
+	}
+
+	for i := 0; i < expected.Len(); i++ {
+		ee := expected.At(i)
+		ae := actual.At(i)
+		errs = multierr.Append(errs, internal.AddErrPrefix(fmt.Sprintf(`event "%s"`, ee.Name()), CompareEvent(ae, ee)))
+	}
+	return errs
+}
+
+// CompareEvent compares each part of two given SpanEvent and returns
+// an error if they don't match. The error describes what didn't match.
+func CompareEvent(expected, actual ptrace.SpanEvent) error {
```

Comment:
Let's call the functions consistently based on the argument type
```suggestion
func CompareSpanEvent(expected, actual ptrace.SpanEvent) error {
```